packages/sdk-auth/src/auth.js

```diff
@@ -106,11 +107,28 @@ export default class SdkAuth {
       config.token || SdkAuth._encodeClientCredentials(credentials)
     const authType = config.authType || constants.DEFAULT_AUTH_TYPE
 
-    let body = `grant_type=${grantType}`
-    if (grantType !== 'refresh_token') body += `&scope=${scope}`
-    if (disableRefreshToken === true) body += '&refresh_token=false'
+    const isNotRefreshTokenGrantType = grantType !== 'refresh_token'
+    const initialBody = encode({
+      grant_type: grantType,
+      ...(disableRefreshToken && { refresh_token: false }),
+      ...(isNotRefreshTokenGrantType && { scope }),
+    })
 
-    return { basicAuth, authType, uri, body, headers }
+    return { basicAuth, authType, uri, body: initialBody, headers }
+  }
+
+  static _appendToRequestBody(
+    request: AuthRequest,
+    toAppend: Object
+  ): AuthRequest {
+    const previousDecodedRequestBody = request.body ? decode(request.body) : {}
+    const nextEncdedRequestBody = encode({
+      ...previousDecodedRequestBody,
+      ...toAppend,
+    })
+    request.body = nextEncdedRequestBody
+
+    return request
```

Comment:
A small helper reused within the module just to append to the request cause that's what we do most of the time here.